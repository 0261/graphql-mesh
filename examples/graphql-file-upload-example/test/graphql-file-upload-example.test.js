const uploadFilesServer = require('../upload-files/server');
const resizeImageServer = require('../resize-image/server');
const { File } = require('cross-undici-fetch')
const { findAndParseConfig } = require('@graphql-mesh/cli');
const { join } = require('path');
const { getMesh } = require('@graphql-mesh/runtime');
const { printSchema, lexicographicSortSchema } = require('graphql');

const mesh$ = findAndParseConfig({
  dir: join(__dirname, '..'),
}).then(config => getMesh(config));

describe('Upload Example', () => {
  let servers = [];
  beforeAll(() => {
    servers.push(
    uploadFilesServer.listen(3001),
    resizeImageServer.listen(3002))
  })
  afterAll(() => {
    servers.forEach(server => server.close())
  });
  it('should generate correct schema', async () => {
    const { schema } = await mesh$;
    expect(
      printSchema(lexicographicSortSchema(schema), {
        descriptions: false,
      })
    ).toMatchSnapshot();
  });
  it('should give correct response', async () => {
    const { execute } = await mesh$;
    const file = new File([Buffer.from('CONTENT')], 'test.txt');
    console.log({
      File,
      file,
      name: file.name
    })
    const result = await execute(
      /* GraphQL */ `
        mutation UploadFile($upload: Upload!) {
          uploadFile(upload: $upload) {
            filename
          }
        }
      `,
      {
        upload: file,
      }
    );
    expect(result?.data?.uploadFile?.filename).toBe('test.txt');
  })
});
