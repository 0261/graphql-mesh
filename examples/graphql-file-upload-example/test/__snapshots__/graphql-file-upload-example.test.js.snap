// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload Example should generate correct schema 1`] = `
"type File {
  base64: String
  filename: String
  thumbnailImage(height: Int, width: Int): String
}

type Mutation {
  deleteFile(filename: String): Boolean
  uploadFile(upload: Upload!): File!
}

type Query {
  files: [File]
  resizeImage(height: Int, image: String, width: Int): String
}

\\"\\"\\"The \`Upload\` scalar type represents a file upload.\\"\\"\\"
scalar Upload"
`;
